_posts/2018-07-25-dota-macos-performance.md

```diff
@@ -0,0 +1,64 @@
+---
+layout: post
+title: Portability benchmark of Dota2 on MacOS
+---
+
+### The Race
+
+gfx-rs is a Rust project aiming to make graphics programming more accessible and portable, with the focus on Vulkan. It's a single Rust API with multiple backends: D3D12/11, Metal, Vulkan, and even OpenGL. We are also building a Vulkan Portability [implementation](https://github.com/gfx-rs/portability) based on it, which allows non-Rust applications using Vulkan to run everywhere.
+
+As our portability implementation was maturing in terms of [CTS](https://github.com/KhronosGroup/VK-GL-CTS) coverage, we started to ask questions about performance. Fortunately, just around this time frame [Valve announced](https://twitter.com/plagman2/status/1002324195135520768?lang=en) an official Metal support for Dota2, via [MoltenVK](https://github.com/KhronosGroup/MoltenVK). We couldn't miss the opportunity to test _gfx-portability_ on Dota2, and replacing MoltenVK appeared to be straightforward. The initial results were... slightly less optimistic. For a while we weren't able to get the game to start, since our Metal backend was missing some required features, like compressed texture support.
```

Comment:
'~~an~~ official Metal support'
'less optimistic' is a bit weird as it isn't obvious what is being compared, maybe disheartening or discouraging?